keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -102,6 +103,12 @@ class ImageService(
         .filter { image ->
           tagsExistForAllAmis(image.tagsByImageId) && image.hasAppVersion
         }
+        // filter to images with the correct base os
+        .filter { image ->
+          // using the xxxxbase pattern means we won't get matches for base OS values that are
+          // substrings of others -- e.g. bionic and bionic-classic
+          baseOs == null || image.baseImageName.startsWith("${baseOs}base")
+        }
```

Comment:
This filters the images returned by River into only those with the specified base OS (if it _was_ specified). Sadly I don't think there's a better way to check the base OS than this. River is extracting base image name and base AMI id from the `description` field in the image itself but I don't see any clearly delimited base OS value in there otherwise I would add a new tag in River.